### PR TITLE
Breakout parsers

### PR DESCRIPTION
**Pull request by tobsan**
Breaking out parsers for most gateways into their own classes, which can be properly unit tested.

### CODE REVIEW

**Review by DunderRoffe — APPROVED** [review]

**Review by DunderRoffe — APPROVED** [review]

File: libsoftwarecontainer/include/gateway.h [hunk 1]
```diff
@@ -154,11 +154,13 @@ class Gateway
     static constexpr const char *ENABLED_FIELD = "enabled";
     static constexpr const char *XDG_RUNTIME_DIR_VARIABLE_NAME = "XDG_RUNTIME_DIR";
```
Comment (breakreturn) [inline libsoftwarecontainer/include/gateway.h]: While at it, could you also see if the defines make sense to have in the base class?

File: libsoftwarecontainer/unit-test/dbusgatewayparser_unittest.cpp [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        "}";
+
+    convertToJSON(config);
+
+    ASSERT_EQ(ReturnCode::SUCCESS,
+              parser.parseDBusConfigElement(configJSON, sessionConf, systemConf));
+    
+    ASSERT_NE(json_array_size(sessionConf), empty);
+    ASSERT_NE(json_array_size(systemConf), empty);
+}
```
Comment (breakreturn) [inline libsoftwarecontainer/unit-test/dbusgatewayparser_unittest.cpp:142]: Please add descriptive comments to all tests. Everywhere ;)

**Review by breakreturn — CHANGES_REQUESTED** [review]

File: libsoftwarecontainer/include/gateway.h (hunk 1 above)
Comment (tobsan) [reply libsoftwarecontainer/include/gateway.h]: Done, it didn't.

File: libsoftwarecontainer/unit-test/dbusgatewayparser_unittest.cpp (hunk 2 above)
Comment (tobsan) [reply libsoftwarecontainer/unit-test/dbusgatewayparser_unittest.cpp:142]: Done!

**Review by breakreturn — APPROVED** [review]

**Pull request merged**